cookbooks/aws-parallelcluster-install/files/default/ec2_udev_rules/manageVolume.py

```diff
@@ -113,40 +97,112 @@ def main():
     # List of available block devices after removing currently used block devices
     available_devices = [a for a in block_devices if a not in paths]
 
-    # Parse configuration file to read proxy settings
-    config = configparser.RawConfigParser()
-    config.read("/etc/boto.cfg")
-    proxy_config = Config()
-    if config.has_option("Boto", "proxy") and config.has_option("Boto", "proxy_port"):
-        proxy = config.get("Boto", "proxy")
-        proxy_port = config.get("Boto", "proxy_port")
-        proxy_config = Config(proxies={"https": "{0}:{1}".format(proxy, proxy_port)})
-
-    # Connect to AWS using boto
-    ec2 = boto3.client("ec2", region_name=region, config=proxy_config)
-
     # Attach the volume
     dev = available_devices[0]
     response = ec2.attach_volume(VolumeId=volume_id, InstanceId=instance_id, Device=dev)
 
     # Poll for volume to attach
     state = response.get("State")
-    x = 0
+    delay = 5  # seconds
+    elapsed = 0
+    timeout = 300  # seconds
     while state != "attached":
-        if x == 60:
-            print("Volume %s failed to mount in 300 seconds." % volume_id)
+        if elapsed == 60:
```

Comment:
I think the logic should be as the following pseudocode, since the meaning of elapsed is still something related to the seconds that passed:

```
delay = 5
elapsed = 0
timeout = 300

while state != "my_desired_state":
    if elapsed >= timeout:
          exit
    do_stuff
    sleep(delay)
    elapsed += delay
```

Reply:
Sounds good, changed